### ISSUE TITLE
Could not build the precompiled application for the device. Error launching application

### ISSUE THREAD

**Issue by SwiftyFlow** [flutter/flutter#131156]
### Is there an existing issue for this?

- [X] I have searched the [existing issues](https://github.com/flutter/flutter/issues)
- [X] I have read the [guide to filing a bug](https://flutter.dev/docs/resources/bug-reports)

### Steps to reproduce

Stop working for 3 months
Wait for Xcode, your iPhone, and Flutter to have updates
Do them all
Try to work
Spend days trying to fix the fact that it "Could not build the precompiled application for the device."
Go on every single threads possible
Flutter upgrade, Flutter downgrade, pub upgrade, pub upgrade --major-versions, downgrade xcode, re-update xcode, ritual dance, voodoo, incantation, prayers, nothing works...

### Actual results

Being stuck for days and about to quit being a developer to reconvert being a painter or to wash cars

### Logs

```
[ +110 ms] executing: sysctl hw.optional.arm64
[  +20 ms] Exit code 1 from: sysctl hw.optional.arm64
[   +1 ms] sysctl: unknown oid 'hw.optional.arm64'
[   +8 ms] executing: [/Users/flowpottevin/Desktop/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +29 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] 135454af32477f815a7525073027a3ff9eff1bfd
[        ] executing: [/Users/flowpottevin/Desktop/flutter/] git tag --points-at 135454af32477f815a7525073027a3ff9eff1bfd
[ +110 ms] Exit code 0 from: git tag --points-at 135454af32477f815a7525073027a3ff9eff1bfd
[        ] 3.3.10
[  +59 ms] executing: [/Users/flowpottevin/Desktop/flutter/] git rev-parse --abbrev-ref --symbolic @{upstream}
[  +19 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{upstream}
[        ] origin/stable
[        ] executing: [/Users/flowpottevin/Desktop/flutter/] git ls-remote --get-url origin
[  +19 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +67 ms] executing: [/Users/flowpottevin/Desktop/flutter/] git rev-parse --abbrev-ref HEAD
[  +21 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] stable
[   +9 ms] executing: sw_vers -productName
[  +24 ms] Exit code 0 from: sw_vers -productName
[        ] macOS
[        ] executing: sw_vers -productVersion
[  +21 ms] Exit code 0 from: sw_vers -productVersion
[        ] 13.4.1
[        ] executing: sw_vers -buildVersion
[  +19 ms] Exit code 0 from: sw_vers -buildVersion
[        ] 22F770820d
[   +5 ms] executing: uname -m
[   +6 ms] Exit code 0 from: uname -m
[        ] x86_64
[  +80 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +5 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +51 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[  +97 ms] executing: which security
[  +36 ms] executing: which openssl
[  +17 ms] executing: security find-identity -p codesigning -v
[ +685 ms]   1) B99BC49EA2A2D22F0F6B751503F6D9E05AB74135 "Developer ID Application: IMS EXCHANGE PTY LTD (D2J9L7NE58)"
                      2) 966C901AD3E756D9BDE65B220A31F3C2EB17E8AD "Developer ID Application: Florent Pottevin (84FL222W2L)"
                      3) 752909094A6D1C759B8AC21DCA9EF9150413862C "Apple Distribution: Florent Pottevin (84FL222W2L)"
                      4) D6BAAB1A9DBBA3330C3F49D4044E122BE73EFBB2 "Apple Development: Florent Pottevin (WUYBS2TTLU)"
                      5) E79BC6C5468B8567D42BA051FD2D791DFB176FF2 "Apple Development: Florent Pottevin (28Y9LLAVCP)"
                         5 valid identities found
[   +4 ms] Found saved certificate choice "Apple Development: Florent Pottevin (WUYBS2TTLU)". To clear, use "flutter config".
[        ] Signing iOS app for device deployment using developer identity: "Apple Development: Florent Pottevin (WUYBS2TTLU)"
[   +1 ms] executing: security find-certificate -c WUYBS2TTLU -p
[  +41 ms] -----BEGIN CERTIFICATE-----
           MIIFoDCCBIigAwIBAgIIK+4H40b+decwDQYJKoZIhvcNAQELBQAwgZYxCzAJBgNV
           BAYTAlVTMRMwEQYDVQQKDApBcHBsZSBJbmMuMSwwKgYDVQQLDCNBcHBsZSBXb3Js
           ZHdpZGUgRGV2ZWxvcGVyIFJlbGF0aW9uczFEMEIGA1UEAww7QXBwbGUgV29ybGR3
           aWRlIERldmVsb3BlciBSZWxhdGlvbnMgQ2VydGlmaWNhdGlvbiBBdXRob3JpdHkw
           HhcNMTcxMDA1MDYwNDIwWhcNMTgxMDA1MDYwNDIwWjCBkzEaMBgGCgmSJomT8ixk
           AQEMCkpZU1dVOTZNOVAxODA2BgNVBAMML2lQaG9uZSBEZXZlbG9wZXI6IEZsb3Jl
           bnQgUG90dGV2aW4gKFdVWUJTMlRUTFUpMRMwEQYDVQQLDAo4NEZMMjIyVzJMMRkw
           FwYDVQQKDBBGbG9yZW50IFBvdHRldmluMQswCQYDVQQGEwJVUzCCASIwDQYJKoZI
           hvcNAQEBBQADggEPADCCAQoCggEBAL/rzGdtgs566IxfwDPAA44lgJqg59AUHfdR
           UfxhtTCqUXRb9xbiVAhKahiloTejM4k/WWDSo1CHkyDd1gBFyqhZ8GPsTsEPNIKG
           pTtv/uD6r/9jWAUjMHzSOePJvlUmBbZAytCemejEHrZ7oymjBgUC0cZJ0nTD8m37
           ClpF44x4twYoaVXkc/bg80x/ePS9srbzVuYu1Pk0c1XXKaku8RYolnSuucnokqqJ
           mQiJMGNaQrT27NlCKwbcaG5rnsz38NgW0a17bsXnM8/DrK89jWPuLInxv5FRhej6
           WV50YjWZnJ5Ue/42L/mYwz9IyrkJeBZD41SS6Gi3ZRuMtNeBngMCAwEAAaOCAfEw
           ggHtMD8GCCsGAQUFBwEBBDMwMTAvBggrBgEFBQcwAYYjaHR0cDovL29jc3AuYXBw
           bGUuY29tL29jc3AwMy13d2RyMDEwHQYDVR0OBBYEFDk56qmjGlBbeFINHtnbqIPR
           v34WMAwGA1UdEwEB/wQCMAAwHwYDVR0jBBgwFoAUiCcXCam2GGCL7Ou69kdZxVJU
           o7cwggEdBgNVHSAEggEUMIIBEDCCAQwGCSqGSIb3Y2QFATCB/jCBwwYIKwYBBQUH
           AgIwgbYMgbNSZWxpYW5jZSBvbiB0aGlzIGNlcnRpZmljYXRlIGJ5IGFueSBwYXJ0
           eSBhc3N1bWVzIGFjY2VwdGFuY2Ugb2YgdGhlIHRoZW4gYXBwbGljYWJsZSBzdGFu
           ZGFyZCB0ZXJtcyBhbmQgY29uZGl0aW9ucyBvZiB1c2UsIGNlcnRpZmljYXRlIHBv
           bGljeSBhbmQgY2VydGlmaWNhdGlvbiBwcmFjdGljZSBzdGF0ZW1lbnRzLjA2Bggr
           BgEFBQcCARYqaHR0cDovL3d3dy5hcHBsZS5jb20vY2VydGlmaWNhdGVhdXRob3Jp
           dHkvMA4GA1UdDwEB/wQEAwIHgDAWBgNVHSUBAf8EDDAKBggrBgEFBQcDAzATBgoq
           hkiG92NkBgECAQH/BAIFADANBgkqhkiG9w0BAQsFAAOCAQEAVDcUucQdxjORRlIZ
           BvwYfJrbv1+Q24VthVahQ2H7xOaQ9D2nJJI6ypZ9gIRmN9LKqW5n5SNLLQxEDFvv
           to+CdTATmUcENTG+237ElmQqEMje3k4JlfC3zrA5AlSItMYK/7GX31mTr5R1wEsL
           EdgDLMQfybkdCmWaG0yZSxusgwgOtM6hgBXn/zWeUpmIWGcJM7o3NYAyaD8Xv2eO
           tj5lPm6HX9Zx0NUC594GHnQN+G9QJpMI9McKeH2bSTsFI0Au/Hwb7JySQfnE8UnR
           IeAbk7wPYEj3O0dG9rXm2VSspa10QeifaSbo1WxfyiXEmof6VESkFX+uAzno/s0t
           Sbd91g==
           -----END CERTIFICATE-----
[   +1 ms] executing: openssl x509 -subject
[  +44 ms] "flutter create" took 1,031ms.
[   +6 ms] "foo." is not a valid Dart package name.
           
           See https://dart.dev/tools/pub/pubspec#name for more information.
[   +1 ms] 
           #0      throwToolExit (package:flutter_tools/src/base/common.dart:10:3)
           #1      CreateBase.projectName (package:flutter_tools/src/commands/create_base.dart:334:9)
           #2      CreateCommand.runCommand (package:flutter_tools/src/commands/create.dart:271:62)
           <asynchronous suspension>
           #3      FlutterCommand.run.<anonymous closure> (package:flutter_tools/src/runner/flutter_command.dart:1209:27)
           <asynchronous suspension>
           #4      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #5      CommandRunner.runCommand (package:args/command_runner.dart:209:13)
           <asynchronous suspension>
           #6      FlutterCommandRunner.runCommand.<anonymous closure> (package:flutter_tools/src/runner/flutter_command_runner.dart:281:9)
           <asynchronous suspension>
           #7      AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #8      FlutterCommandRunner.runCommand (package:flutter_tools/src/runner/flutter_command_runner.dart:229:5)
           <asynchronous suspension>
           #9      run.<anonymous closure>.<anonymous closure> (package:flutter_tools/runner.dart:62:9)
           <asynchronous suspension>
           #10     AppContext.run.<anonymous closure> (package:flutter_tools/src/base/context.dart:150:19)
           <asynchronous suspension>
           #11     main (package:flutter_tools/executable.dart:91:3)
           <asynchronous suspension>
           
           
[ +102 ms] ensureAnalyticsSent: 101ms
[   +1 ms] Running shutdown hooks
[        ] Shutdown hooks complete
[        ] exiting with code 1
``` 

### Flutter Doctor output

```
[✓] Flutter (Channel stable, 3.3.10, on macOS 13.4.1 22F770820d darwin-x64, locale en-AU)
    • Flutter version 3.3.10 on channel stable at /Users/flowpottevin/Desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 135454af32 (7 months ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc1)
    • Android SDK at /Users/flowpottevin/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14E300c
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

[✓] VS Code (version 1.80.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.68.0

[✓] Connected device (2 available)
    • iFlow 14 Pro (mobile) • 00008120-000E7CA01E03C01E • ios            • iOS 16.5.1 20F75
    • Chrome (web)          • chrome                    • web-javascript • Google Chrome 114.0.5735.198

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

**Comment by dam-ease** [comment]
Hi @SwiftyFlow. Thanks for filing this issue. Can you help provide more information on the reproduction steps. More specifically:

- Have you been able to successfully update Xcode?
- Do you get this log when you run the `flutter upgrade` command? If not, can you share the logs from `flutter upgrade` and also share what command you ran that gave these logs?
- You might want to uninstall and re-install flutter from scratch again to see if this issue reproduces.

Thank you.

**Comment by SwiftyFlow** [comment]
- xcode is up to date
- logs from flutter upgrade:

Flutter 3.10.6 • channel stable • https://github.com/flutter/flutter.git
Framework • revision f468f3366c (12 days ago) • 2023-07-12 15:19:05 -0700
Engine • revision cdbeda788a
Tools • Dart 3.0.6 • DevTools 2.23.1

Running flutter doctor...
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.4.1 22F770820d darwin-x64, locale en-AU)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.1)
[✓] VS Code (version 1.80.1)
[✓] Connected device (1 available)
[✓] Network resources

• No issues found!

**Comment by dam-ease** [comment]
Thanks for your response @SwiftyFlow. Everything seems fine with the logs from flutter upgrade. Where/When do you get this error?

**Comment by SwiftyFlow** [comment]
I don't know what has changed since Sunday but I now can install the app on my iPhone - problem solved after re-doing flutter upgrade + changed intl to ^0.18.0 then this has pub upgrade(d) and then it worked. However I noticed things done during the build that were not done on Sunday (thin build I don't know what and 2 more steps that now have been erased from the log as I already restarted the app...) Anyway closing issue now. Thanks!
